### PR TITLE
Fixes false positives for Helm test authentication validation

### LINKED ISSUE #312: Authentication validation Helm test always passes (false positives)
## Is your feature request related to a problem? Please describe.

The current Helm tests are always returning success for the Conjur authentication validation test case. 

Some other changes that should be made:

- The validation tests for Conjur account and authenticator ID:
   - Should be skipped if the Conjur URL is not reachable
   - Should check for authorization errors
    - Should decode YAML from the /info endpoint before checking for robustness.
       This can be done in bash using `yq` to ensure we're comparing against the proper info field.
- The checks for error codes in the authenticator logs should be skipped if authentication is successful.

## Describe the solution you would like

See above.

### PR DESCRIPTION
**Pull request by diverdane**
### What does this PR do?

The current Helm tests are always returning success for the Conjur
authentication validation test case. This change fixes this issue so
the the proper test result is reported.

Some other changes that are included:
- The validation tests for Conjur account and authenticator ID are changed
  in the following ways:
  - Test cases are skipped if the Conjur URL is not reachable
  - Test cases check for authorization errors
  - Test cases decode the YAML returned from the /info endpoint
    using `yq` to ensure we're comparing against the proper info field.
- The checks for error codes in the authenticator logs are change in the
  following ways:
  - Error checks are not performed if authentication is successful
  - New test case added for error code CAKC007

### What ticket does this PR close?
Resolves #312 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [x] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

#### Manual tests
**If you are preparing for a release**, have you run the following manual tests to verify existing functionality continues to function as expected?
- [ ] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local authn-k8s client image build

### CODE REVIEW

**Review by rpothier — DISMISSED** [review]
Nice cleanup!

File: helm/kubernetes-cluster-prep/templates/tests/_helm_test.bats.txt [hunk 1]
```diff
@@ -22,32 +20,120 @@ readonly AUTHN_TIMEOUT_SECS=5
...
+
+###################################################
+#  Validation Tests Not Requiring Authentication  #
+###################################################
+
 @test "Conjur Appliance URL is a reachable address" {
-  cmd=(curl -k --connect-timeout 5 "$conjurApplianceUrl":443)
-  display_info "Running ${cmd[@]}"
-  run "${cmd[@]}"
+  display_info "Attempting to reach Conjur URL with 'curl -k ...'"
```
[the remark this answers is not in the document]
Comment (diverdane) [reply helm/kubernetes-cluster-prep/templates/tests/_helm_test.bats.txt:49]: I debated on leaving the display of the actual curl command... but the command was moved up to the `conjur_is_reachable` helper function, so display of the curl command would have to be added there. I decided not to add it there for readability and to keep the helper function very low level. In the end, I think we're providing the user enough high level info to know what the issue is.

File: helm/kubernetes-cluster-prep/templates/tests/_helm_test.bats.txt [hunk 2]
```diff
@@ -22,32 +20,120 @@ readonly AUTHN_TIMEOUT_SECS=5
...
+###################################################
+
 {{- if .Values.test.authentication.enable }}
 
 @test "Conjur authenticator sidecar has successfully retrieved an API token" {
   display_info "Checking for existence of access token at '$ACCESS_TOKEN_FILE'"
   secs=0
-  until [ "$secs" -ge "$AUTHN_TIMEOUT_SECS" ]
   status=0
+  until [ "$secs" -ge "$AUTHN_TIMEOUT_SECS" ]
```
[the remark this answers is not in the document]
Comment (diverdane) [reply helm/kubernetes-cluster-prep/templates/tests/_helm_test.bats.txt:130]: The problem was that the `status=0` was being done between the `until` and the `do` statements:
```
until [ "$secs" -ge "$AUTHN_TIMEOUT_SECS" ]
status=0
do
```
This was apparently causing the `status=0` to get executed on the last iteration of `until`, meaning it was wiping out the status from the last iteration of `run`.

In fact, we don't actually need the `status=0` initialization. I'll delete that to make it cleaner.

**Review by rpothier — APPROVED** [review]
LGTM

**Pull request merged**